core-plugins/src/main/java/co/cask/hydrator/plugin/common/FileSetUtil.java

```diff
@@ -0,0 +1,157 @@
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.hydrator.common.HiveSchemaConverter;
+import co.cask.hydrator.common.batch.JobUtils;
+import com.google.common.base.Throwables;
+import org.apache.avro.Schema;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.Job;
+import parquet.avro.AvroParquetInputFormat;
+import parquet.avro.AvroParquetOutputFormat;
+
+import java.io.IOException;
+import java.util.Map;
+
+
+/**
+ * Utilities for configuring file sets during pipeline configuration.
+ * TODO (CDAP-6211): Why do we lower-case the schema for Parquet but not for Avro?
+ */
+public class FileSetUtil {
+
+  /**
+   * Configure a file set to use Parquet file format with a given schema. The schema is lower-cased, parsed
+   * as an Avro schema, validated and converted into a Hive schema. The fileset is configured to use
+   * Parquet input and output format, and also configured for Explore to use Parquet. The schema is added
+   * to the file set properties in all the different required ways:
+   * <ul>
+   *   <li>As a top-level dataset property;</li>
+   *   <li>As the schema for the input and output format;</li>
+   *   <li>As the schema of the Hive table.</li>
+   * </ul>
+   * @param configuredSchema the original schema configured for the table
+   * @param properties a builder for the file set properties
+   */
+  public static void configureParquetFileSet(String configuredSchema, FileSetProperties.Builder properties) {
+
+    // validate and parse schema as Avro, and attempt to convert it into a Hive schema
+    String lowerCaseSchema = configuredSchema.toLowerCase();
+    Schema avroSchema = parseAvroSchema(lowerCaseSchema, configuredSchema);
+    String hiveSchema = parseHiveSchema(lowerCaseSchema, configuredSchema);
+
+    properties
+      .setInputFormat(AvroParquetInputFormat.class)
+      .setOutputFormat(AvroParquetOutputFormat.class)
+      .setEnableExploreOnCreate(true)
+      .setExploreFormat("parquet")
+      .setExploreSchema(hiveSchema.substring(1, hiveSchema.length() - 1))
+      .add(DatasetProperties.SCHEMA, lowerCaseSchema);
+
+    Job job = createJobForConfiguration();
+    Configuration hConf = job.getConfiguration();
+    AvroParquetInputFormat.setAvroReadSchema(job, avroSchema);
+    for (Map.Entry<String, String> entry : hConf) {
+      properties.setInputProperty(entry.getKey(), entry.getValue());
+    }
+    hConf.clear();
+    AvroParquetOutputFormat.setSchema(job, avroSchema);
+    for (Map.Entry<String, String> entry : hConf) {
+      properties.setOutputProperty(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * Configure a file set to use Avro file format with a given schema. The schema is parsed
+   * as an Avro schema, validated and converted into a Hive schema. The fileset is configured to use
```

Comment:
configured to use Avro key input and output format. 

There happens to be a proper name after the line break, but the sentence continues.